### PR TITLE
Comdb2 plugin framework

### PR DESCRIPTION
**Pull request by nirbhayc**
Plugin framework
- Framework to build/install static or dynamic plugins
- Interfaces to define/install appsock and opcode handlers
- Move newsql appsock handler to plugin/newsql
- Cleanup legacy stuff

System tables
- comdb2_plugins
- comdb2_appsock_handlers
- comdb2_opcode_handlers

### CODE REVIEW

File: db/appsock_handler.c [hunk 1]
```diff
@@ -892,46 +364,22 @@ void appsock_handler_start(struct dbenv *dbenv, SBUF2 *sb)
         logmsg(LOGMSG_ERROR, "%s:thdpool_enqueue error\n", __func__);
         net_end_appsock(sb);
         return;
-    } 
+    }
 
     total_appsock_conns++;
-    LOCK(&appsock_conn_lk) { active_appsock_conns++; }
+    LOCK(&appsock_conn_lk)
+    {
```
Comment (adizaimi) [inline db/appsock_handler.c]: Looks like we can get rid of this lock and just use atomic_add for this one.

File: plugin/newsql/newsql.c [hunk 2]
```diff
@@ -0,0 +1,970 @@
...
+        cdb2__query__free_unpacked(query, &pb_alloc);
+        return NULL;
+    }
+#endif
+    return query;
+}
+
+extern int gbl_allow_incoherent_sql;
+
+int handle_newsql_request(comdb2_appsock_arg_t *arg)
```
Comment (adizaimi) [inline plugin/newsql/newsql.c]: Does this need to be a plugin? I thought this one did not need to be a plugin.

**Comment by nirbhayc** [comment]
Fix for cdb2api.test : [PR-661](https://github.com/bloomberg/comdb2/pull/661)

**Comment by riverszhang89** [comment]
Can you please replace `printf/fprintf` in `db/sltdbt.c` and `db/appsock_handler.c` with `logmsg`?

**Pull request merged**